src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs

```diff
@@ -12644,5 +12644,26 @@ void M(Guid g1, Guid g2)
                 LanguageVersion = LanguageVersion.CSharp10,
             }.RunAsync();
         }
+
+        [WorkItem(58804, "https://github.com/dotnet/roslyn/issues/58804")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryCast)]
+        public async Task ConvertingMethodGroupToObject()
+        {
+            var code = @"
+class C
+{
+    static object M(object o)
+    {
+        return (object)o.ToString;
+    }
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp10,
+            }.RunAsync();
+        }
```

Comment:
please add test where the cast *is* removed if the return type is `Delegate`, `MulticastDelegate`, or things like `Func<string>`.  thanks!

Reply:
just to verify, we have tests for these three cases?

Reply:
Not for all the three I think. I'll get to adding the tests soon.

Reply:
https://github.com/dotnet/roslyn/blob/bd1a3814c5eaf3da54c45f80bf4b57062a8d6e9d/src/Analyzers/CSharp/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.cs#L1954-L1970

@CyrusNajmabadi Is this expected for this test?

Reply:
@CyrusNajmabadi This scenario is now broken (regression by this PR):

```csharp
using System;

class C
{
    static Delegate M(object o)
    {
        return (Delegate)(object)o.ToString;
    }
}
```

Is this the scenario you were looking for?